### DISCUSSION TITLE
Why does bind_visibility_from not work on my plotly element?

### DISCUSSION THREAD

**Discussion by duane-space** [zauberzeug/nicegui#1064] [Q&A] [answered]
### Question

In the below (hopefully at least close to minimally reproducible) code example, I would expect the plotly element to start off as not visible, and then become visible after clicking the button for the first time. Actual behavior is that it is still not visible after adding rows to the table.

The plot works as expected if I don't include the bind_visibility at all, and even works as written if there is a row in the table when it is first created (meaning the plotly element is always visible).

Thanks for any help!

```py
import plotly.express as px
import pandas as pd
from nicegui import ui


def add_row(table):
    table.add_rows({'name': 'Alice', 'age': 18})


df = pd.DataFrame([
    dict(Start='2023-06-22T14:00:00', Finish='2023-06-22T14:10:00', Label="label"),
])

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]

table = ui.table(columns=columns, rows=[], row_key='name')

fig = px.timeline(df, x_start="Start", x_end="Finish", y="Label")

plot = ui.plotly(fig).bind_visibility_from(table, "rows", lambda x: len(x) > 0)

ui.button('add a row',on_click=lambda: add_row(table))

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Oh, thanks for spotting this, @duane-impulse!

It definitely looks like a bug. Here is a minimal reproduction:

```py
plot = ui.plotly(go.Figure(go.Scatter(x=[], y=[])))
ui.button('hide', on_click=lambda: plot.set_visibility(False)). # should hide the plot
```

I'll open an issue for tracking this bugfix.

**Reply by falkoschindler** [reply]
As a workaround you can call `super(ui.plotly, plot).update()` to manually send the update to the client.

**Reply by duane-space** [reply]
I see this fix made it in to 1.2.23, thanks!
